src/applications/vaos/new-appointment/newAppointmentFlow.js

```diff
@@ -161,18 +170,22 @@ export default {
     async next(state, dispatch) {
       const data = getFormData(state);
       const communityCareEnabled = vaosCommunityCare(state);
+      const useVSP = vaosVSPAppointmentNew(state);
 
       // check that the result does have a ccId
       if (getTypeOfCare(data)?.ccId !== undefined) {
         try {
           if (communityCareEnabled) {
             // Check if user registered systems support community care...
-            const userSystemIds = selectSystemIds(state);
-            const ccSites = await getSitesSupportingVAR(userSystemIds);
-            const ccEnabledSystems = userSystemIds.filter(id =>
-              ccSites.some(site => site.id === id),
+            const siteIds = selectSystemIds(state);
+            const parentIds = await getOrganizations({ siteIds, useVSP });
+            const ccSites = await getSitesSupportingVAR(parentIds);
+            const ccEnabledSystems = parentIds.filter(parent =>
+              ccSites.some(
+                site => site.id === getSiteIdFromOrganization(parent),
+              ),
             );
-            dispatch(updateCCEnabledSystems(ccEnabledSystems));
+            dispatch(updateCCEnabledSystems(ccEnabledSystems, parentIds));
```

Comment:
I think it was originally @jenniemc that copy pasted these lines from `typeOfCare`'s `next()` but maybe we could use this as a chance to copy this into a common function such as `getCCEnabledSystems()` since it seems to be doing the same exact thing as lines 94-102.